libs/enclave/src/lib/scoring/feature-extractor.ts

```
@@ -0,0 +1,482 @@
+
+    // Parse code into AST
+    const ast = this.parseCode(code);
+
+    // Extract tool calls and patterns
+    const toolCalls: ExtractedToolCall[] = [];
+    const toolSequence: string[] = [];
+    const sensitiveFields = new Set<string>();
+    const sensitiveCategories = new Set<SensitiveCategory>();
+
+    // Track loop depth during traversal
+    let loopDepth = 0;
```

ESLint

[error] 'loopDepth' is never reassigned. Use 'const' instead.

(prefer-const)